packages/file-search/src/browser/quick-file-open.ts

```diff
@@ -126,79 +130,76 @@ export class QuickFileOpenService implements QuickOpenModel {
         this.cancelIndicator.cancel();
         this.cancelIndicator = new CancellationTokenSource();
         const token = this.cancelIndicator.token;
-        const proposed = new Set<string>();
+        
         const rootUri = workspaceFolder.uri;
-        const handler = async (result: string[]) => {
-            if (!token.isCancellationRequested) {
-                const root = new URI(rootUri);
-                result.forEach(p => {
-                    const uri = root.withPath(root.path.join(p)).toString();
-                    proposed.add(uri);
-                });
-                const itemPromises = Array.from(proposed).map(uri => this.toItem(uri));
-                acceptor(await Promise.all(itemPromises));
+        const root = new URI(rootUri);
+        const alreadyCollected = new Set<string>();
+        
+        const recentlyUsedItems: QuickOpenItem[] = [];
+        const locations = this.navigationLocationservice.locations()
+        if (locations.length > 0) {
```

Comment:
No need for this `if`.

Reply:
Why not?

Reply:
One line below, you iterate over the items, why is it necessary to check the length first? If it is `0` then it will not run into the for-loop, right?

Reply:
Of course :). Didn't see it in the diff.